src/neural/network_opencl.cc

```diff
@@ -59,72 +59,97 @@ class OpenCLComputation : public NetworkComputation {
  public:
   OpenCLComputation(const OpenCL_Network& opencl_net,
                     const OpenCLWeights& weights)
-      : opencl_net_(opencl_net),
-        weights_(weights),
-        input_data_(kInputPlanes * 64),
-        value_data_(weights_.num_value_channels),
-        policy_data_(),
-        q_value_(0) {}
+      : opencl_net_(opencl_net), weights_(weights), policies_(), q_values_() {}
 
   virtual ~OpenCLComputation() {}
 
   // Adds a sample to the batch.
   void AddInput(InputPlanes&& input) override { planes_.emplace_back(input); }
 
- public:
   // Do the computation.
   void ComputeBlocking() override {
-    for (auto& sample : planes_) ComputeBlocking(sample);
-  }
+    // Determine the largest batch for allocations.
+    const auto plane_count = planes_.size();
+    const auto max_batch_size = opencl_net_.getMaxMatchSize();
+    const auto largest_batch_size = std::min(max_batch_size, plane_count);
 
-  void ComputeBlocking(const InputPlanes& sample) {
-    int index = 0;
-    for (const InputPlane& plane : sample) {
-      float value = plane.value;
-      const uint64_t one = 1;
-      for (int i = 0; i < 64; i++)
-        input_data_[index++] = (plane.mask & (one << i)) != 0 ? value : 0;
-    }
+    const auto num_output_policies = weights_.num_output_policies;
+    const auto num_value_channels = weights_.num_value_channels;
+
+    /* Typically
```

Comment:
We use `//` style comments, not `/* */`.

Reply:
Maybe that should start with the license statement ?